### ISSUE TITLE
Best way to write @When I go to the :arg1 node with the title :arg2 in Behat 3?

### ISSUE THREAD

**Issue by sagannotcarl** [jhedstrom/drupalextension#451]
I am porting an old project from Behat 2 to 3, and I have a lot of steps where I want to visit a node that was either explicitly created in a previous step or was generated automatically in a previous step. 

I used to have `$this->nodeLoadByTitle($type, $title)` helper function in my FeatureContext that used a `db_select` query to find the node I'm looking for but in my new setup I get an error: `Call to undefined function db_select`.

Two questions:
- Is there a better way of doing this? 
- If my custom helper function is the right way to do this, why is `db_select` undefined now but it wasn't before?

Thanks for any pointers!
Colin

**Comment by jhedstrom** [comment]
Hmm, `db_select` should be available if Drupal is bootstrapped. Are you loading the proper contexts in `behat.yml`? Something like:

```yaml
default:
  suites:
    default:
      contexts:
        - FeatureContext
        - Drupal\DrupalExtension\Context\ConfigContext
        - Drupal\DrupalExtension\Context\DrupalContext
        - Drupal\DrupalExtension\Context\MessageContext
        - Drupal\DrupalExtension\Context\MinkContext
        - Drupal\DrupalExtension\Context\MarkupContext
```

**Comment by sagannotcarl** [comment]
@jhedstrom: Thanks for the help, I didn't have all of those contexts loaded. But unfortunately I'm still getting the error `Call to undefined function db_select()`. 

Here is my new entire `behat.yml`:

```
default:
  suites:
    default:
      contexts:
        - FeatureContext
        - Drupal\DrupalExtension\Context\ConfigContext
        - Drupal\DrupalExtension\Context\DrupalContext
        - Drupal\DrupalExtension\Context\MessageContext
        - Drupal\DrupalExtension\Context\MinkContext
        - Drupal\DrupalExtension\Context\MarkupContext
        - Drupal\DrupalExtension\Context\DrushContext
  extensions:
    Behat\MinkExtension:
      goutte: ~
      selenium2: ~
      base_url: http://nginx
    Drupal\DrupalExtension:
      blackbox: ~
      api_driver: "drush"
      drush:
       root: ../.
      text:
        username_field: "Email"
```

Also, all of my behat files are inside of a `test` directory, do you think it would work better if I installed everything in my root? I'm using Drupal 7, but I'm re-setting up behat from scratch.

**Comment by jhedstrom** [comment]
Does your `FeatureContext` extend the `RawDrupalContext`? Before calling `db_select`, can you try running `$this->getDriver()->bootstrap();?` If it's not extending that, you can use a `@BeforeScenario` hook to inject the `DrupalContext` class into yours, something like this:

```php
  /**
   * Before scenario setup.
   *
   * @param BeforeScenarioScope $scope
   *   Hook containing Behat environment, feature, and scenario.
   *
   * @BeforeScenario
   */
  public function gatherContexts(BeforeScenarioScope $scope) {
    $environment = $scope->getEnvironment();
    $this->drupalContext = $environment->getContext('Drupal\DrupalExtension\Context\DrupalContext');
 }
```

Then you can call `$this->drupalContext->getDriver()->bootstrap();`.

**Comment by jhedstrom** [comment]
Actually, this may be your issue: ` api_driver: "drush"`. That won't actually bootstrap drupal directly, as it assumes all calls are going through drush itself (which handles connecting to Drupal). If you are running tests on the same machine as the Drupal installation, you can switch that to `drupal` instead, and then the code above should get you going...

**Comment by sagannotcarl** [comment]
My `FeatureContext` extend the `RawDrupalContext`, and I tried adding `$this->getDriver()->bootstrap();` into my helper function but still getting `undefined`.

I had previously tried `api_driver: "drush"` and never got any scenarios to run. No matter what I would get the error `No specifications found at path(s) 'features'. This might be because of incorrect paths configuration in your 'suites'.`

I tried all sorts of different options for the paths configuration but nothing worked. I will try re-installing behat in my root drupal directory (right now it's in a `tests` subdirectory), maybe that will help.
